cli/container.go

```diff
@@ -9,37 +9,53 @@ import (
 	"os"
 	"time"
 
-	"github.com/docker/go-connections/nat"
-
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/api/types/network"
 	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
 )
 
-func createServer(verbose bool, image string, port string, args []string, env []string, name string, volumes []string) (string, error) {
-	log.Printf("Creating server using %s...\n", image)
+func startContainer(verbose bool, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, containerName string) (string, error) {
 	ctx := context.Background()
+
 	docker, err := client.NewEnvClient()
 	if err != nil {
 		return "", fmt.Errorf("ERROR: couldn't create docker client\n%+v", err)
 	}
-	reader, err := docker.ImagePull(ctx, image, types.ImagePullOptions{})
-	if err != nil {
-		return "", fmt.Errorf("ERROR: couldn't pull image %s\n%+v", image, err)
-	}
-	if verbose {
-		_, err := io.Copy(os.Stdout, reader)
+
+	resp, err := docker.ContainerCreate(ctx, config, hostConfig, networkingConfig, containerName)
```

Comment:
@ibuildthecloud I found the problem with this to be the following:
If you create a cluster without having the image pulled already, this will return an empty `resp`, thus line 48 ([docker.ContainerStart](https://github.com/rancher/k3d/pull/27/files#diff-794246841b704a4567b6a716f52c9134R49)) will fail with `page not found` response from the docker daemon.
So either we check for the existence of the image in some different way first or we call the `ContainerCreate` once again in the conditional body :+1:

Reply:
oops, sorry forgot to call create again, fixing it right now

Reply:
:+1: